restapi/api_object.go

```diff
@@ -239,7 +258,17 @@ func (obj *api_object) read_object() error {
 		return err
 	}
 
-	err = obj.update_state(res_str)
+	if obj.read_search {
+		obj.search_path = strings.Replace(obj.get_path, "{id}", obj.id, -1)
+		res_str_fltr, err := obj.filter_object(res_str, obj.search_key, obj.search_value, obj.results_key)
```

Comment:
As noted below, I think this could just be a call to [`find_object`](https://github.com/Mastercard/terraform-provider-restapi/blob/master/restapi/api_object.go#L285). To work for this use case, which requires a string, the `find_object` function should be amended to return the object (and the error, if any)... and then we can marshal it to a string in this small branch of the code.

I propose modifying `find_object` like so:
`func (obj *api_object) find_object(query_string string, search_key string, search_value string, results_key string) *map[string]interface{}, error {`

So, somewhere in the beginning of the function, declare a pointer to a `map[string]interface{}` which will be our return object, but only set it [when the object is found around line 368](https://github.com/Mastercard/terraform-provider-restapi/blob/master/restapi/api_object.go#L368).